definitions.tex

```diff
@@ -12,5 +12,123 @@
 structures (e.g. ASTs, metadata, etc) or any intermediate representation chosen
 by the implementer.
 
+\indexdefn{module}
+\indexdefn{module|named}
+\definition{module, named module}{definitions.module}
+The collection of translation units whose \grammarterm{module-declaration}
+specify the same \grammarterm{module-name}.
+See \Cpp{} Standard [module.unit] paragraph 1.
+
+\indexdefn{module!unit}
+\definition{module unit}{definitions.module-unit}
+A translation unit that contains a \grammarterm{module-declaration}.
+See \Cpp{} Standard [module.unit] paragraph 1.
+
+\indexdefn{module!interface}
+\definition{module interface}{definitions.module-interface}
+The set of declarations belonging to a module that are necessarily reachable
+(\Cpp{} Standard [module.reach] paragraph 1) from outside of that module.
+See \Cpp{} Standard [module].
+
+\indexdefn{module!interface unit}
+\definition{module interface unit}{definitions.module-interface-unit}
+A module unit whose \grammarterm{module-declaration} starts with
+\grammarterm{export-keyword}.
+See \Cpp{} Standard [module.unit] paragraph 2.
+
+\indexdefn{module!implementation unit}
+\definition{module implementation unit}{definitions.module-implementation-unit}
+A module unit which is not a module interface unit.
+See \Cpp{} Standard [module.unit] paragraph 2.
```

Comment:
From @Bigcheese on the 2020-05-22 telecon:

This doesn't really say anything. Explain what this actually is.

It's important that we express that a module unit is either an interface unit or an implementation unit.

Consider the examples from [module.unit]/4.